src/parser.rs

```diff
@@ -849,6 +851,56 @@ impl<'a> Parser<'a> {
         })
     }
 
+    pub fn parse_ceil_expr(&mut self) -> Result<Expr, ParserError> {
+        self.expect_token(&Token::LParen)?;
+        let expr = self.parse_expr()?;
+        let keyword_to = self.expect_keyword(Keyword::TO);
+        match keyword_to {
+            Err(_) => {
+                // Parse `CEIL(x)`
+                self.expect_token(&Token::RParen)?;
+                Ok(Expr::Ceil {
+                    expr: Box::new(expr),
+                    field: DateTimeField::NoDateTime,
+                })
+            }
+            _ => {
+                // Parse `CEIL(expr TO DateTimeField)`
+                let field = self.parse_date_time_field()?;
+                self.expect_token(&Token::RParen)?;
+                Ok(Expr::Ceil {
+                    expr: Box::new(expr),
+                    field,
+                })
+            }
+        }
+    }
+
+    pub fn parse_floor_expr(&mut self) -> Result<Expr, ParserError> {
```

Comment:
This function is almost the same as `parse_ceil_expr` -- would it be possible to refactor them and reduce the duplication?